mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+    EvaluationMetrics,
+    EvaluationArtifact,
+)
+from mlflow.utils.file_utils import TempDir
+from mlflow.tracking.artifact_utils import get_artifact_uri
+from sklearn import metrics as sk_metrics
+import matplotlib.pyplot as pyplot
+import scikitplot
+import shap
+import math
+
+shap.initjs()
+
+"""
+[P0] Accuracy: Calculates how often predictions equal labels.
+[P0] BinaryCrossentropy: Computes the crossentropy metric between the labels and predictions.
+[P0] Hinge: Computes the hinge metric between y_true and y_pred.
+[P0] Sum: Computes the (weighted) sum of the given values.
+[P0] Mean: Computes the (weighted) mean of the given values.
+[P0] ExampleCount: Computes the total number of evaluation examples.
+[P0] MeanAbsoluteError: Computes the mean absolute error between the labels and predictions.
+[P0] MeanSquaredError: Computes the mean squared error between y_true and y_pred.
+[P0] RootMeanSquaredError: Computes root mean squared error metric between y_true and y_pred.
+
+[P0] TrueNegatives: Calculates the number of true negatives.
+[P0] TruePositives: Calculates the number of true positives.
+[P0] FalseNegatives: Calculates the number of false negatives.
+[P0] FalsePositives: Calculates the number of false positives.
+https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix
+
+[P0] Precision: Computes the precision of the predictions with respect to the labels.
+[P0] Recall: Computes the recall of the predictions with respect to the labels.
+[P0] AUC: Approximates the AUC (Area under the curve) of the ROC or PR curves.
+[P0] F1 Score: 2*precision*recall / (precision+recall)
+
+[P0] BinaryClassConfusionMatrix
+
+Plots
+[P0] Confusion matrix
+[P0] Interactive ROC curve with metrics (TP/TN/FP/FN/Acc/F1/AUC), binary classification
+[P0] Lift chart
+
+Global explainability
+[P0] Model built-in feature importance (supported models)
+[P0] SHAP explainers
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+
+    @classmethod
+    def save_content_to_file(cls, content, output_artifact_path):
+        assert isinstance(content, Image)
+        content.save(output_artifact_path)
+
+    @classmethod
+    def load_content_from_file(cls, local_artifact_path):
+        return open_image(local_artifact_path)
+
+
+class DefaultEvaluator(ModelEvaluator):
+
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_image_artifact(self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name):
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        mlflow.log_artifact(artifact_file_local_path, artifact_file_name)
```

Comment:
can we use `mlflow.log_figure`?

Reply:
The demo notebook you shared seems to save the file within a directory of the same name, for example:

<img width="392" alt="Screen Shot 2021-11-24 at 4 48 28 PM" src="https://user-images.githubusercontent.com/39497902/143332182-c67ccbc6-b076-4349-9310-7c45bf8a1867.png">

Can we fix this to remove the nesting?